changelog/std-file-createTempFile.dd

```diff
@@ -0,0 +1,15 @@
+`createTempFile` added to std.file
+
+$(REF std, file, createTempFile) creates a file in $(REF tempDir, std, file)
+with a random name (with an optionally provided prefix and suffix) and returns
+the file name. It also optionally takes data to write to the file when creating
+it (if no data is provided, then the newly created file will be empty).
```

Comment:
It's awkward that the file is created, closed, to presumably be opened by the user again right after. Shouldn't this function return e.g. an opened `File`? Then people can call `name` to get its name.

Reply:
Returning an open `File` would fit in std.stdio, not std.file, because std.file specifically only ever deals with entire files at at time. Also, in most cases, I'd personally find it very annoying to have a `File` to deal with. The main purpose is to write data to a file for testing or to pass to another program and then pass the file name to something else to open it. Having an open `File` rather than just a filename just makes that use case more annoying. We did have a version of this in std.stdio briefly that did that, but it got removed from std.stdio on the grounds that it increased the size of "hello world" due to issues with the compiler.

The way I would expect to use this normally would be to just call this function like `write` and write the whole thing at once, and in the rarer case where they want a `File`, they can just not pass any data to it and then reopen it with `File` (in which case, since the file was already created, there is no security risk like there is the first time).